src/scripts/main.js

```diff
@@ -1,3 +1,34 @@
 'use strict';
 
-// write code here
+function getNumber(str) {
+  return +str.split('').filter(num => !isNaN(num)).join('');
+}
+
+function sortList(list) {
+  return list.sort((a, b) => (
+    getNumber(b.dataset.salary) - getNumber(a.dataset.salary)));
+}
+
+function getEmployees(list) {
+  const employeesInfo = [];
+
+  list.forEach(item => (
+    employeesInfo.push({
+      name: item.textContent.trim(),
+      position: item.dataset.position,
+      salary: item.dataset.salary,
+      age: item.dataset.age,
+    })
+  ));
+
+  return employeesInfo;
+}
+
+const employees = [ ...document.querySelectorAll('li') ];
+
+sortList(employees);
+getEmployees(employees);
```

Comment:
Why don’t you catch the result getEmployees function? Maybe should use this result for rendering a new list on the page?